### PR TITLE
handle IO::WaitWritable

### PR DESCRIPTION
**Pull request by nerdrew**
cc @zanker @mdayaram @tarcieri

### CODE REVIEW

**Comment by nerdrew** [comment]
related comments: https://github.com/redis/redis-rb/pull/496#pullrequestreview-6319252

**Comment by mdayaram** [comment]
Tested this patch in our app and it solved our issue 👍

**Comment by tarcieri** [comment]
Yeah this looks like an oversight in the original implementation. 👍  on the patch

File: lib/redis/connection/ruby.rb [hunk 1]
```diff
@@ -27,8 +27,12 @@ module SocketMixin
       CRLF = "\r\n".freeze
 
       # Exceptions raised during non-blocking I/O ops that require retrying the op
-      NBIO_EXCEPTIONS = [Errno::EWOULDBLOCK, Errno::EAGAIN]
-      NBIO_EXCEPTIONS << IO::WaitReadable if RUBY_VERSION >= "1.9.3"
+      NBIO_READ_EXCEPTIONS = [Errno::EWOULDBLOCK, Errno::EAGAIN]
+      NBIO_WRITE_EXCEPTIONS = [Errno::EWOULDBLOCK, Errno::EAGAIN]
+      if RUBY_VERSION >= "1.9.3"
```
Comment (tarcieri) [inline lib/redis/connection/ruby.rb]: Sidebar: any plans for deprecating Ruby 1.8 support? It's been EOL for 2 years now...

Comment (badboy) [inline lib/redis/connection/ruby.rb]: In the next major release, if we ever find the time to get to it.

**Comment by nerdrew** [comment]
Ping? Thoughts on this PR?

**Comment by mdayaram** [comment]
Thanks guys!

BTW, @djanowski would it be possible to release a version 3.3.2 with this fix?

**Comment by mdayaram** [comment]
@djanowski any idea when 3.3.2 will be cut? we just got bit by an issue related to this again.

**Comment by djanowski** [comment]
It's out.

**Comment by mdayaram** [comment]
thanks!

**Comment by quixoten** [comment]
This change introduced Timeout errors in our environment. I have a fix up in #649. It should not compromise this fix, but I'd appreciate additional :eyes: on it to confirm.

**Pull request merged**